substrate/frame/assets/src/tests.rs

```diff
@@ -1775,3 +1775,79 @@ fn asset_destroy_refund_existence_deposit() {
 		assert_eq!(Balances::reserved_balance(&admin), 0);
 	});
 }
+
+#[cfg(all(feature = "try-runtime", test))]
+#[test]
+fn migrate_to_v2_works() {
+	new_test_ext().execute_with(|| {
+		use crate::migration::v2::old as v1;
+		use frame_support::{pallet_prelude::StorageVersion, traits::OnRuntimeUpgrade};
+		use sp_runtime::traits::Get;
+		StorageVersion::new(1).put::<Pallet<Test, ()>>();
+
+		// Create an asset with id 1
+		v1::Asset::<Test, ()>::insert(
+			1,
+			v1::AssetDetails::<
+				<Test as Config>::Balance,
+				<Test as frame_system::Config>::AccountId,
+				DepositBalanceOf<Test>,
+			> {
+				owner: 1,
+				issuer: 1,
+				admin: 1,
+				freezer: 1,
+				supply: 102,
+				deposit: 0,
+				min_balance: 1,
+				is_sufficient: false,
+				accounts: 0,
+				sufficients: 0,
+				approvals: 0,
+				status: AssetStatus::Live,
+			},
+		);
+
+		// Populate balances of asset 1
+		Account::<Test, ()>::insert(
+			1,
+			1,
+			AssetAccountOf::<Test, ()> {
+				balance: 100,
+				status: AccountStatus::Liquid,
+				reason: ExistenceReason::<_, _>::Sufficient,
+				extra: Default::default(),
+			},
+		);
+		Account::<Test, ()>::insert(
+			1,
+			2,
+			AssetAccountOf::<Test, ()> {
+				balance: 2,
+				status: AccountStatus::Liquid,
+				reason: ExistenceReason::<_, _>::Sufficient,
+				extra: Default::default(),
+			},
+		);
+
+		// Define a constant Vec for migration.
+		// This is a workaround for the fact that we can't implement Get for Vec.
+		struct ConstVecForMigrationToV2;
+		impl Get<Vec<(u32, u64)>> for ConstVecForMigrationToV2 {
+			fn get() -> Vec<(u32, u64)> {
+				// AssetId 1, AccountId 2 will be inactive.
+				vec![(1, 2)]
+			}
+		}
```

Comment:
Can you just define a constant vector in the migrations file the same as it would be defined in the runtime?

Reply:
also how do we calculate what this vec should be when we go to run this for real?

Reply:
This pattern is not something I came up with, rather a solution I found in [balances](https://github.com/paritytech/polkadot-sdk/blob/c54ea64af43b522d23bfabb8d917a490c0f23217/substrate/frame/balances/src/migration.rs#L68), introduced by @gavofyork in [Substrate PR 12832](https://github.com/paritytech/substrate/pull/12832), and used again later in [society](https://github.com/paritytech/polkadot-sdk/blob/c54ea64af43b522d23bfabb8d917a490c0f23217/substrate/frame/society/src/migrations.rs#L38), introduced in [Substrate PR 11324](https://github.com/paritytech/substrate/pull/11324) - not sure on which one was written first, since society was merged later but was opened before.

In the latter, I see the migration has an internal method `from_original`, which is the one called in the [test](https://github.com/paritytech/polkadot-sdk/blob/c54ea64af43b522d23bfabb8d917a490c0f23217/substrate/frame/society/src/tests.rs#L80) instead of building a struct as I did. Should I proceed with the same approach? 

I can try to search how this pattern is used in runtimes if needed.

Reply:
I may be missing something but I'm a bit confused actually why any accounts will need to be seeded with inactive balance when we merge this PR.

Isn't the only way their bal can be made inactive is by calling `deactivate`, which we are introducing in this PR?

Reply:
When introducing the inactive balance feature, one might want to populate the state during the migration, by passing a list of accounts which balances are to be considered inactive.

On pallet balances I found:
> // Generally this will be used for the XCM teleport checking account.

Reply:
@KiChjang does xcm need this?

Reply:
XCM needs this but I don't think Assets does, at least to the extent the pallet is used in Asset Hub. AH doesn't trust any other chains as teleporters for its locally created assets, and for foreign assets it does not maintain a checking account (it expects the asset creator to).

- [Local Assets uses checking](https://github.com/polkadot-fellows/runtimes/blob/dcb9f62/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L105), 
    - [but no teleporters](https://github.com/polkadot-fellows/runtimes/blob/dcb9f62/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L406)
- [Foreign Assets no checking](https://github.com/polkadot-fellows/runtimes/blob/dcb9f62/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L136)

Reply:
Thanks @joepetrowski , so don't need to implement seeding inactive balances?